Updates to MRS Notebook1 based on review

drlaw1558 opened the pull request:
New PR addressing some of the issues discussed by https://github.com/spacetelescope/jwebbinar_prep/pull/45

ojustino commented:
Hello @drlaw1558:

### Before you begin

I've pushed my attempt at running your notebook. To view and edit it locally, follow these steps:

```
git checkout drljun17
git fetch YOUR-FORK-REMOTE
git merge YOUR-FORK-REMOTE/drljun17
```

_(`YOUR-FORK-REMOTE` is your fork's online copy. It's often `origin`, but if  you don't know your name for it, run `git remote -a` and choose the one whose path ends with `git@github.com:YOURUSERNAME/jwebbinar_prep.git`.)_

From here you can work on your branch as normal. If you have trouble with this step, please let me know before continuing.

### Specific comments (\*action required\*)

Some of these remarks are repeated from the previous review. To expedite the merging process, please ensure you've addressed all comments before pushing your new commit(s).

1. **I'm still unable to run the notebook from top to bottom.**
    - I understand that you've included a link to a Box directory, but **the link needs to be integrated into the code to simplify the process of running the notebook**. (If you're not sure how, see the "Download the data" section of [`nirspec_fs.ipynb`](https://github.com/spacetelescope/jwebbinar_prep/blob/main/spec_mode/nirspec_fs.ipynb) for an example.) Under the current construction...
        - ...the `redolong = False` case cannot work because the user will lack access to the cached files.
        - ...`mirisim_dir = 'stage0/'` cannot work unless that directory is present. Users who clone this repository currently only get subdirectories `stage1/`, `stage2/`, and `stage3/` in this notebook's folder. As such, cells that depend on the presence of `stage0/` will fail.
        - ...the default value for `cache_dir` needs to be changed to the "on your own machine" case; the `/home/shared/preloaded-fits/mrs-data/notebook1/` path is specific to the JWebbinar's Jupyter Hub environment and won't work for users who clone the repository.
2. **There are still several PEP8 errors.** Here is a way to address them:
    1. Run `pip install flake8 pycodestyle_magic` on your command line to install [`flake8`](https://github.com/mattijn/pycodestyle_magic).
    2. Restart the notebook.
    3. Run imports and then the [cells I added](https://user-images.githubusercontent.com/12895749/121743209-fd7f8a80-cace-11eb-86a5-90e7b857a8be.png) beneath them.
    4. **Edit each code cell that shows style warnings [like this](https://user-images.githubusercontent.com/12895749/121729210-306c5300-cabc-11eb-90eb-eb494dca53c4.png)**, re-run the cell, and repeat until all style warnings have been addressed. I will leave a comment on an exemplary cell since you said you were having trouble parsing my commit from #45.
    5. Delete the cells I added and clear notebook outputs before committing and pushing.
3. Other comments:
    - Looking at the other JWebbinar folders, your notebooks would be the only ones that are hidden within multiple subdirectories. If you insist that it's necessary to break the pattern, I would at least **give the extra directory a more descriptive name than `mirimrs`** so that users understand there are more notebooks to see here.
    - The ReviewNB output is going to be lengthy because the notebook has quite a few cells and most of them have errors since the issues I described with `mirisim_dir` and `cache_dir` happen near the beginning. Besides the warning/error outputs and the cells labeled "Reviewer note," no other changes were made to the code.

ojustino commented on spec_mode/mirimrs/notebook1/MRS_Notebook1.ipynb:
```diff
@@ -24,7 +24,7 @@
    "source": [
```
<p>This cell can probably go now that you've added the <code>jwst</code> version check earlier on in the notebook.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/spacetelescope/jwebbinar_prep/pull/46/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

ojustino replied:
<p>This is an example of a cell with several PEP8 warnings printed as output below the cell. (The <code>IndexError</code> is a separate issue that should be fixed once the notebook runs in full.) In this case, every warning is for a location where there's no space after a comma -- index 56 of line 11, index 14 of line 10, etc. Remember that the preferred style is <code>a, b, c</code> instead of <code>a,b,c</code>.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/spacetelescope/jwebbinar_prep/pull/46/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='59'/>

ojustino replied:
<p>This cell can probably go now that you've added the <code>jwst</code> version check earlier on in the notebook.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/spacetelescope/jwebbinar_prep/pull/46/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

ojustino replied:
<p>This is an example of a cell with several PEP8 warnings printed as output below the cell. (The <code>IndexError</code> is a separate issue that should be fixed once the notebook runs in full.) In this case, every warning is for a location where there's no space after a comma -- index 56 of line 11, index 14 of line 10, etc. Remember that the preferred style is <code>a, b, c</code> instead of <code>a,b,c</code>.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/spacetelescope/jwebbinar_prep/pull/46/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

drlaw1558 commented:
Per slack discussion, I've moved the Box import to a jwebbinar path, and added a red note to the notebook that the data need to be obtained offline.  Also deleted MRSNotebook2 from the spec_mode directory since Webbinar #5 chose to put this in the ifu_session instead and we don't need two copies.  With this removed the need for directories to keep outputs from notebooks 1/2 separate are no longer needed, so moved Notebook1 higher up the directory tree.
